### PR TITLE
fix(k8s): handle 404 exception when tagging image for deletion

### PR DESCRIPTION
**Pull request by solomonope**
**What this PR does / why we need it**:
This PR allows a 404 error when tagging images in the registry for deletion fail silently and log a message

**Which issue(s) this PR fixes**:
#1472
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts [hunk 1]
```diff
@@ -169,20 +169,24 @@ async function deleteImagesFromRegistry(ctx: KubernetesPluginContext, log: LogEn
...
-    // Issue the delete request
-    await queryRegistry(ctx, log, `${name}/manifests/${digest}`, {
-      method: "DELETE",
-    })
+      // Issue the delete request
+      await queryRegistry(ctx, log, `${name}/manifests/${digest}`, {
+        method: "DELETE",
+      })
+    } catch (err) {
+      log.debug(`${err.message} happend when deleting ${image}`)
```
Comment (edvald) [inline garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: I think we need to more specifically catch the error. If you just wrap the first query and explicitly catch a 404, you can just return without logging any error. Other errors should be thrown as usual.

Comment (solomonope) [reply garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: I have made the change, I have updated to only check for 404

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts [hunk 2]
```diff
@@ -169,20 +169,27 @@ async function deleteImagesFromRegistry(ctx: KubernetesPluginContext, log: LogEn
...
-    // Issue the delete request
-    await queryRegistry(ctx, log, `${name}/manifests/${digest}`, {
-      method: "DELETE",
-    })
+      // Issue the delete request
+      await queryRegistry(ctx, log, `${name}/manifests/${digest}`, {
+        method: "DELETE",
+      })
+    } catch (err) {
+      if (err.response && !(err.response.status === 404)) {
```
Comment (edvald) [inline garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: Minor nitpick, but `err.response.status !== 404` is a bit cleaner :)

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts (hunk 1 above)
Comment (edvald) [reply garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: That works. I'd just remove that log line, don't think it's necessary. Then it's good to go!

Comment (solomonope) [reply garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: fixed

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts (hunk 2 above)
Comment (solomonope) [reply garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: fixed

**Pull request merged**